Semantic Scholar.js

```diff
@@ -75,128 +60,55 @@ function doWeb(doc, url) {
-			|| rawData.primaryPaperLink.linkType == 'arxiv')) {
-			item.attachments.push({
-				url: rawData.primaryPaperLink.url,
-				title: "Full Text PDF",
-				mimeType: 'application/pdf'
-			});
-			if (rawData.primaryPaperLink.linkType == 'arxiv') {
-				let arxivId = rawData.primaryPaperLink.url.match(/\d{4}\.\d{5}/);
-				if (arxivId.length >= 1) {
-					if (item.extra) {
-						item.extra += '\narXiv: ' + arxivId[0];
-					}
-					else {
-						item.extra = 'arXiv: ' + arxivId[0];
-					}
-				}
-			}
-		}
+	let type = citation.split('{')[0].replace('@', '');
+	const itemType = bibtex2zoteroTypeMap[type];
 
-		if (rawData.paperAbstract && rawData.paperAbstract.text) {
-			item.abstractNote = ZU.unescapeHTML(rawData.paperAbstract.text);
-		}
+	var item = new Zotero.Item(itemType);
 
-		if (rawData.doiInfo && rawData.doiInfo.doi) {
-			item.DOI = rawData.doiInfo.doi;
-		}
-		
-		if (rawData.entities) {
-			for (let entity of rawData.entities) {
-				item.tags.push(entity.name);
-			}
-		}
+	// load structured schema data
+	const schemaTag = doc.querySelector("script.schema-data");
+	const schemaObject = JSON.parse(schemaTag.innerHTML);
+	const article = schemaObject["@graph"][1][0];
 
-		item.complete();
-	});
-	translator.translate();
-}
+	item.title = article.name;
+	item.abstractNote = article.description;
 
-// Some page ranges are given as e.g. 575-84. Expand these to e.g. 575-584
-function fixPageRange(pageRange) {
-	let numbers = pageRange.split('-');
-	if (numbers.length !== 2) {
-		return pageRange;
+	if (article.author) {
+		article.author.forEach((author) => {
+			item.creators.push(ZU.cleanAuthor(author.name, 'author'));
+		});
 	}
-	
-	numbers = numbers.map(function (x) {
-		return parseInt(x);
+	item.publicationTitle = article.publication;
+	item.date = article.datePublished;
+
+	// attachments
+	item.attachments.push({
+		url: url,
+		title: "Semantic Scholar Link",
+		mimeType: "text/html",
+		snapshot: false
 	});
-	
-	// No change is needed if they're already correctly formatted
-	if (numbers[0] < numbers[1]) {
-		return pageRange;
+
+	const paperLink = article.about.url;
+	if (paperLink.includes("pdfs.semanticscholar.org") || paperLink.includes("arxiv.org")) {
+		item.attachments.push({
+			url: paperLink,
+			title: "Full Text PDF",
+			mimeType: 'application/pdf'
+		});
 	}
 	else {
-		let digitsInSecond = Math.floor(Math.log10(numbers[1])) + 1;
-		let baseNumber = numbers[0];
-		let difference = 0;
-		
-		for (let i = 1; i <= digitsInSecond; i++) {
-			let mod = baseNumber % (10 ** i);
-			baseNumber -= mod;
-			difference += mod;
-		}
-		
-		// If the given pageRange doesn't make sense, just leave it as it has been given
-		// e.g. '95-10'
-		if (difference > numbers[1]) {
-			return pageRange;
-		}
-		
-		numbers[1] = baseNumber + numbers[1];
-		
-		return numbers[0] + '-' + numbers[1];
+		item.attachments.push({
+			url: paperLink,
+			title: "Publisher Link",
+			mimeType: "text/html",
+			snapshot: false
+		});
```

Comment:
If this is just a doi.org URL, it doesn't need to be saved, since the DOI field alone is sufficient to resolve it.

If it's an actual publisher URL — i.e., the thing that might be get cited when a style doesn't prefer the DOI — it should be saved to the parent item's URL field instead. Linked-URL attachments are just for secondary sources (e.g., Semantic Scholar page, Amazon product page, etc.).

Reply:
It can be a variety of things, I'm going to pull this part out for now (have added the DOI more reliably below)